lib/esbuild.js

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env node
+
+const esbuild = require('esbuild');
+const postcss = require('esbuild-postcss');
+const browserslistToEsbuild = require('browserslist-to-esbuild');
+const babel = require('esbuild-plugin-babel');
+const { resolve } = require('path');
+const fs = require('fs');
+
+const isDev = process.env.NODE_ENV === 'development';
+
+const defaultBuild = {
+  outdir: 'public/dist',
+  entryNames: '[name].[hash]',
+  minify: !isDev,
+  loader: {
+    '.png': 'file',
+    '.svg': 'file',
+    '.woff2': 'file'
+  },
+  metafile: true,
+  bundle: true,
+  target: browserslistToEsbuild(),
+  plugins: isDev
+    ? [
+        postcss()
+      ]
+    : [
+        postcss(),
+        babel({
+          filter: /.*js/,
+          namespace: '',
+          config: {} // babel config here or in babel.config.json
+        })
+      ],
+  watch: isDev
+    ? {
+      onRebuild(error, result) {
+        if (error) {
+          console.error('ESBuild rebuilt failed:', error);
+        } else {
+          console.log('ESBuild watch rebuilt');
+          makeAssetsMap(result.metafile.outputs);
+        }
+      },
+    }
+    : false,
+};
+
+const assetsBuild = JSON.parse(fs.readFileSync(resolve('assets.json'), 'utf8'));
+
+function makeAssetsMap(outputs) {
+  const config = require(`${process.cwd()}/app/config`);
+  const assetOpts = config.engine.opts.assets;
+
+  // console.log(outputs);
+  const map = {};
+
+  Object.entries(outputs).forEach(([path, value]) => {
+    const key = value.entryPoint
+      ? resolve(value.entryPoint)
+      : resolve(Object.keys(value.inputs)[0]);
+
+    const hashedPath = path.replace(/^public/, '');
+
+    map[key] = {};
+
+    if (assetOpts.inline.includes(key)) {
+      const cssPath = resolve(__base, '../public', hashedPath.replace(/^\//, ''));
+      map[key].content = fs.readFileSync(cssPath, 'utf8');
+    }
+
+    map[key].path = key.match(/\.svg$/)
+      ? hashedPath
+      : `${assetOpts.origin}${hashedPath}`;
+  });
+
+  // console.log(map);
```

Comment:
Delete line

Reply:
Done. I thought you'd want to keep those in for debugging, which is why I left them.